src/app/product-list/product-list.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Product } from '../model/product';
+
+@Component({
+  selector: 'app-product-list',
+  templateUrl: './product-list.component.html',
+  styleUrls: ['./product-list.component.scss']
+})
+export class ProductListComponent {
+  @Input() productList!: Product[];
+  @Output() addProduct = new EventEmitter();
+  @Output() editProduct: EventEmitter<Product> = new EventEmitter<Product>();
+  @Output() deleteProduct: EventEmitter<number> = new EventEmitter<number>();
+
+  displayedColumns: string[] = ['id', 'name', 'category', 'quantity', 'price', 'delete'];
+
+  constructor() { }
+
+  updateProduct(product: Product) {
+    this.editProduct.emit({...product});
```

Comment:
If you need to break the reference, that should be done in the parent component since it's not intrinsic to the product list components functionality. The goal is to have your child display components be as light weight as possible.